java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaEndpointNavigationProvider.java

```diff
@@ -97,6 +101,33 @@ private void buildEndpointAnnotations(@NotNull SearchScope searchScope) {
         });
     }
 
+    @Nullable
+    protected PsiFile getPsiFile(@NotNull VirtualFile virtualFile) {
+        final PsiFile psiFile;
+        try {
+            psiFile = PsiUtils.uriToPsiFile(virtualFile.getUrl(), project);
+        } catch (PsiFileNotFountException e) {
+            // very unlikely
+            return null;
+        }
+        return psiFile;
+    }
+
+    private void buildEndpointAnnotations(@NotNull VirtualFile virtualFile) {
+        final PsiFile psiFile = getPsiFile(virtualFile);
+        if (psiFile == null) return;
+
```

Comment:
if psiFile is null then something is wrong with the whole logic.
the logic of discovery should only start when we have a valid java psi file.

Reply:
what do you suggest, fail fast, and throw exception?

Reply:
OK, I agree its unlikely.  maybe just a log message so we have an idea when that happens. it may happen on corrupted index of if the file was deleted.